io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -1336,6 +1391,60 @@ pub async fn nexus_create_v2(
     }
 }
 
+async fn get_nexus_zone_info_from_children(children_devices: &mut Vec<(String, String)>) -> Result<ZoneInfo, Error> {
+    let mut zone_info = ZoneInfo::default();
+    let mut conventional = false;
+    let mut info_set = false;
+
+    for (_uri, device_name) in &*children_devices {
+        let dev = device_lookup(&device_name).unwrap();
+
+        if dev.is_zoned() {
+            zone_info.zoned = true;
+            if !info_set {
+                zone_info.num_zones = dev.get_num_zones();
+                zone_info.zone_size = dev.get_zone_size();
+                zone_info.max_zone_append_size = dev.get_max_zone_append_size();
+                zone_info.max_open_zones = dev.get_max_open_zones();
+                zone_info.max_active_zones = dev.get_max_active_zones();
+                zone_info.optimal_open_zones = dev.get_optimal_open_zones();
+                info_set = true;
+            } else if zone_info.num_zones != dev.get_num_zones()
+                    || zone_info.zone_size != dev.get_zone_size()
+                    || zone_info.max_zone_append_size != dev.get_max_zone_append_size()
+                    || zone_info.max_open_zones != dev.get_max_open_zones()
+                    || zone_info.max_active_zones != dev.get_max_active_zones()
+                    || zone_info.optimal_open_zones != dev.get_optimal_open_zones() {
+                error!("Can not use ZBD's with different parameters as nexus children");
+                return Err(Error::MixedZonedChild { child: device_name.to_string() });
+           }
+        } else {
+            conventional = true;
+        }
+
+        if zone_info.zoned == conventional {
+            error!("Can not handle conventional and zoned storage at the same time in a nexus");
```

Comment:
maybe include the nexus in the log too?

```suggestion
            error!("{nexus:?} - can not handle conventional and zoned storage at the same time in a nexus");
```

Reply:
The nexus does not exist at this point, but I can include the name of the requested nexus.